build-scripts/build_llvm.py

```diff
@@ -23,9 +23,15 @@ def clone_llvm(dst_dir, llvm_repo, llvm_branch):
 
     if not llvm_dir.exists():
         GIT_CLONE_CMD = f"git clone --depth 1 --branch {llvm_branch} {llvm_repo} llvm"
-        print(GIT_CLONE_CMD)
+        print(f"cd {dst_dir} && {GIT_CLONE_CMD}")
         subprocess.check_output(shlex.split(GIT_CLONE_CMD), cwd=dst_dir)
 
+        patch_dir = pathlib.Path(__file__).parent.joinpath("llvm-patches").resolve()
```

Comment:
i suspect this should be target dependent.

while we are currently using the similar version (llvm-18) for all targets, it has not always been the case.
and it will unlikely be the case in future.
cf. https://github.com/bytecodealliance/wasm-micro-runtime/pull/4654